File: tests/machine/x/pl/cross_join_triple.pl
:- style_check(-singleton).
get_item(Container, Key, Val) :-
    is_dict(Container), !, (string(Key) -> atom_string(A, Key) ; A = Key), get_dict(A, Container, Val).
get_item(Container, Index, Val) :-
    string(Container), !, string_chars(Container, Chars), nth0(Index, Chars, Val).
get_item(List, Index, Val) :- nth0(Index, List, Val).

:- initialization(main, main).
main :-
    Nums = [1, 2],
    Letters = ["A", "B"],
    Bools = [true, false],
    dict_create(_V0, map, [N-N, L-L, B-B]),
    findall(_V0, (member(N, Nums), member(L, Letters), member(B, Bools), true), _V1),
    Combos = _V1,
    writeln("--- Cross Join of three lists ---"),
    catch(
        (
            member(C, Combos),
                catch(
                    (
                        get_item(C, 'n', _V2),
                        writeln(_V2),
                        true
                    ), continue, true),
                    fail
                ; true
            ), break, true),
            true.

File: tests/machine/x/fs/cross_join_triple.fs
open System

type Anon1 = {
    n: int
    l: string
    b: bool
}
let nums: int list = [1; 2]
let letters: string list = ["A"; "B"]
let bools: bool list = [true; false]
let combos: obj list = [ for n in nums do 
  for l in letters do 
  for b in bools do yield { n = n; l = l; b = b } ]
printfn "%s" "--- Cross Join of three lists ---"
for c in combos do
    printfn "%s" (String.concat " " [string c.n; string c.l; string c.b])
